[Flaky Test] ExampleWrapper - `Log in goroutine after TestDurationIsAddedToEvent has completed`

## Flaky Test

* **Test Name:** ExampleWrapper
* **Link:** https://github.com/elastic/beats/blob/a682c1d831242fffe21c4cb6357b5e7cd8636b7d/metricbeat/mb/module/example_test.go#L40
* **Branch:** `main`
* **Artifact Link:** https://github.com/elastic/beats/actions/runs/19643631806/job/56253342645?pr=46587#step:3:1222
* **Notes:** 

### Stack Trace

```
=== Failed
=== FAIL: metricbeat/mb/module ExampleWrapper (unknown)
=== RUN   ExampleWrapper
panic: Log in goroutine after TestDurationIsAddedToEvent has completed: 2025-11-24T17:56:32.748Z	DEBUG	module	Stopped Wrapper[name=fake, len(metricSetWrappers)=1]
```

--- a/metricbeat/mb/module/wrapper_test.go
+++ b/metricbeat/mb/module/wrapper_test.go
@@ -224,8 +224,6 @@ func TestDurationIsAddedToEvent(t *testing.T) {
 	require.NoError(t, err)
 
 	done := make(chan struct{})
-	defer close(done)
-
 	output := m.Start(done)
 
 	event := <-output
@@ -233,6 +231,13 @@ func TestDurationIsAddedToEvent(t *testing.T) {
 	fields := event.Fields.Flatten()
 	assert.Contains(t, fields, "event.duration", "event.duration should be present in event")
 	assert.Greater(t, fields["event.duration"], time.Duration(0), "event.duration should be greater than 0")
+
+	// stop worker
+	close(done)
+
+	// wait for shutdown to prevent logging after test completes
+	event, ok := <-output
+	assert.Falsef(t, ok, "received unexpected event: %+v", event)
 }
 
 func TestNewWrapperForMetricSet(t *testing.T) {

